src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,168 @@
+
+import jakarta.validation.Valid;
+
+import java.time.LocalDateTime;
+
+@Tag(name = "RecommendationRequest")
+@RequestMapping("/api/recommendationrequest")
+@RestController
+@Slf4j
+public class RecommendationRequestController extends ApiController {
+    @Autowired
+    RecommendationRequestRepository recommendationRequestRepository;
+
+    /**
+     * List all recommendation requests
+     * 
+     * @return an iterable of RecommendationRequest
+     */
+    @Operation(summary= "List all recommendation requests")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<RecommendationRequest> allRecommendationRequests() {
+        Iterable<RecommendationRequest> requests = recommendationRequestRepository.findAll();
+        return requests;
+    }
+
+    /**
+     * Create a new request
+     * 
+     * @param requesterEmail   
+     * @param requesterName     
+     * @param professorEmail  
+     * @param professorName
+     * @param requestType      
+     * @param details           
+     * @param submissionDate
+     * @param completionDate
+     * @param status
+     * @return                  
+     */
+    @Operation(summary= "Create a new request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequest(
+            @Parameter(name="requesterEmail") @RequestParam String requesterEmail,
+            @Parameter(name="requesterName") @RequestParam String requesterName,
+            @Parameter(name="professorEmail") @RequestParam String professorEmail,
+            @Parameter(name="professorName") @RequestParam String professorName,
+            @Parameter(name="requestType") @RequestParam String requestType,
+            @Parameter(name="details") @RequestParam String details,
+            @Parameter(name="submissionDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("submissionDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime submissionDate,
+            @Parameter(name="completionDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("completionDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime completionDate,
+            @Parameter(name="status") @RequestParam String status)
+            throws JsonProcessingException {
+
+        // For an explanation of @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+        // See: https://www.baeldung.com/spring-date-parameters
+
+        // log.info("dateRequested={}", dateRequested);
+        // log.info("dateNeeded={}", dateNeeded);
+
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+        recommendationRequest.setRequesterEmail(requesterEmail);
+        recommendationRequest.setRequesterName(requesterName);
```

Comment:
Why is there still a requesterEmail and requesterName field?  This should be replaced with requesterId.